contracts/solver-registry/src/admin.rs

```diff
@@ -40,6 +48,69 @@ impl Contract {
     #[access_control_any(roles(Role::Owner))]
     pub fn add_full_access_key(&mut self, public_key: PublicKey) -> Promise {
         assert_one_yocto();
+        Event::FullAccessKeyAdded {
+            public_key: &public_key,
+        }
+        .emit();
         Promise::new(env::current_account_id()).add_full_access_key(public_key)
     }
+
+    /// Withdraw tokens from a pool to the receiver account id.
+    #[payable]
+    #[access_control_any(roles(Role::Owner))]
+    pub fn withdraw_from_pool(
+        &mut self,
+        pool_id: u32,
+        receiver_id: AccountId,
+        token_id: AccountId,
+        amount: U128,
+    ) -> PromiseOrValue<U128> {
+        assert_one_yocto();
+        let pool = self.pools.get(pool_id).expect("Pool not found");
```

Comment:
```suggestion
        let pool = self.pools.get(pool_id).unwrap_or_else(|_| env::panic_str(ERR_POOL_NOT_FOUND));
```

Nit: should use Near SDK panic instead of Rust native panic; and can reuse the error message defined in `pool.rs`.